actors/datacap/src/lib.rs

```diff
@@ -416,29 +413,59 @@ where
         };
         Ok(rec)
     }
+}
+
+impl<'a, RT> Syscalls for &Messenger<'a, RT>
+where
+    RT: Runtime,
+{
+    fn root(&self) -> Result<Cid, NoStateError> {
+        self.rt.get_state_root().map_err(|_| NoStateError)
+    }
+
+    fn receiver(&self) -> ActorID {
+        self.rt.message().receiver().id().unwrap()
+    }
+
+    // TODO: What's the best way to avoid this duplication?
```

Comment:
Help sought here.

Reply:
The solution now looks good, please remove this comment.